tests/e2e/gc_rwx_multi_ns_gc.go

```diff
@@ -0,0 +1,286 @@
+		scParameters      map[string]string
+		storagePolicyName string
+		volHealthCheck    bool
+	)
+	ginkgo.BeforeEach(func() {
+		client = f.ClientSet
+		// TODO: Read value from command line
+		volHealthCheck = true
+		namespace = getNamespaceToRunTests(f)
+		scParameters = make(map[string]string)
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+		bootstrap()
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+	})
+	/*
+		Verify static volume provisioning works across Namespace in the same GC
+		1. Create a SC
+		2. Create a PVC with "ReadWriteMany" access mode in namespace-1 using SC
+		3. Wait for PVC to be in the Bound phase
+		4. Verify CnsVolumeMetadata CRD is created
+		5. Verify if the mapping PVC is bound in the SV cluster using the volume handler
+		6. Verify health status of PVC
+		7. Verify volume is created on CNS by using CNSQuery API and also check metadata is pushed to CNS
+		8. Change reclaimPolicy of PV to Retain
+		9. Delete PVC from namespace-1
+		10. Verify CnsVolumeMetadata CRD is deleted
+		11. Verify PVC name is removed from volume entry on CNS
+		12. Create PV2 with "ReadWriteMany" access mode statically backed by the same VolumeID (file share) from PV1
+		13. Create PVC2 with "ReadWriteMany" access mode which gets bounds to PV2 in namespace-2
+		14. Wait for PVC to be in the Bound phase
+		15. Verify CnsVolumeMetadata CRD is created
+		16. Verify if the mapping PVC is bound in the SV cluster using the volume handler
+		17. Verify health status of PVC
+		18. Verify volume is created on CNS by using CNSQuery API and also check metadata is pushed to CNS
+		19. Create a Pod1 using PVC2 created above at a mount path specified in PodSpec
+		20. Verify CnsFileAccessConfig CRD is created
+		21. Verify Pod1 is in the Running phase
+		22. Verify ACL net permission set by calling CNSQuery for the file volume
+		23. Create a file (file1.txt) at the mount path. Check if the creation is successful
+		24. Delete Pod1
+		25. Verify CnsFileAccessConfig CRD is deleted
+		26. Verify if the Pod is successfully deleted
+		27. Delete PVC in GC
+		28. Verify if PVC is deleted in GC
+		29. Delete PVs
+		30. Verify if PVs are deleted in the SV cluster and GC
+		31. Verify CnsVolumeMetadata CRD is deleted
+		32. Check if the VolumeID is deleted from CNS by using CNSQuery API
+	*/
+	ginkgo.It("[rwm-csi-tkg] Verify static volume provisioning works across Namespace in the same GC", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		var storageclasspvc *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var pv *v1.PersistentVolume
+		var err error
+
+		ginkgo.By("CNS_TEST: Running for GC setup")
+		ginkgo.By("Creating namespace on second GC")
```

Comment:
where are you fetching the client for second GC
I see all actions going to only one GC here

Reply:
It is a second namespace in the same GC - corrected the log message